Fix Board Analog API methods (#3002)
viamrobotics/docs: docs/components/board/_index.md

### Read

> Read the current value from the analog reader.

Reviewer: We should update the return of that analog Read function it now returns the following message in value: https://github.com/viamrobotics/api/blob/4f9d14d067b9b7965fcb4e89d6185e9dc348c3cd/proto/viam/component/board/v1/board.proto#L191

So it is not a struct, not one integer,  the sturct returns are as value.Value (integer), value.Min (float) value.Max (float), value.StepSize (float)

Author: Ok thanks!

**Returns:**

> - [(Analog.Value)](https://python.viam.dev/autoapi/viam/components/board/index.html#viam.components.board.Board.Analog.Value): The current value, including the integer `value` of the digital signal output by the analog pin and the `min_range`, `max_range`, and `step_size` of the reader.

Reviewer: the python docs for value put it succinctly 

> The result of reading an analog reader. It contains the raw data read, the reader’s minimum and maximum possible values, and its step size (the minimum possible change between values it can read)

> - [(Analog.Value)](https://python.viam.dev/autoapi/viam/components/board/index.html#viam.components.board.Board.Analog.Value): The current value, including the integer `value` of the digital signal output by the analog pin and the `min_range`, `max_range`, and `step_size` of the reader.

Author: ```suggestion
- [(Analog.Value)](https://python.viam.dev/autoapi/viam/components/board/index.html#viam.components.board.Board.Analog.Value): The result of reading an analog reader. It contains the raw data read as `value`, the reader’s minimum and maximum possible values as `min_range` and `max_range`, and its `step_size` (the minimum possible change between values it can read).
```

> - [(Analog.Value)](https://python.viam.dev/autoapi/viam/components/board/index.html#viam.components.board.Board.Analog.Value): The result of reading an analog reader. It contains the raw data read as `value`, the reader’s minimum and maximum possible values as `min_range` and `max_range`, and its `step_size` (the minimum possible change between values it can read).

Author: @randhid Is this ok or do you want me to use precise sentence from upstream docs?

Reviewer: very ok

> - [(int)](https://docs.python.org/3/library/functions.html#int): The value of the digital signal output by the analog pin.

Reviewer: Add the new returned messages here.

Author: Did so for python and go!

reading, err := analog.Read(context.Background(), nil)
readingValue := reading.Value
stepSize := reading.StepSize
```

{{% /tab %}}
{{< /tabs >}}

### Write

Write an analog value to a pin or analog output on a board capable of doing so.

{{< tabs >}}
{{% tab name="Python" %}}

**Parameters:**

- `value` [(int)](https://docs.python.org/3/library/functions.html#int): Value to write to the pin.
- `extra` [(Optional\[Dict\[str, Any\]\])](https://docs.python.org/library/typing.html#typing.Optional): Extra options to pass to the underlying RPC call.
- `timeout` [(Optional\[float\])](https://docs.python.org/library/typing.html#typing.Optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.

**Returns:**

- None

For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/board/client/index.html#viam.components.board.client.AnalogClient.write).

```python {class="line-numbers linkable-line-numbers"}
my_board = Board.from_robot(robot=robot, name="my_board")

# Get the AnalogWriter “my_example_analog_writer”.

> writer = await my_board.analog_by_name(name=”my_example_analog_writer”)

Author: ```suggestion
writer = await my_board.analog_by_name(name="my_example_analog_writer")
```

// Get the value of the digital signal "my_example_analog" has most recently measured.
reading := analog.Read(context.Background(), nil)
reading_value := reading.Value
```

{{% /tab %}}
{{< /tabs >}}

### Write

> Write a value to the analog writer.

Author: thanks! @randhid

{{% /tab %}}
{{< /tabs >}}

### Write

Write a value to the analog writer.

{{< tabs >}}
{{% tab name="Python" %}}

**Parameters:**

- `value` [(int)](https://docs.python.org/3/library/functions.html#int): Value to write to the pin.
- `extra` [(Optional\[Dict\[str, Any\]\])](https://docs.python.org/library/typing.html#typing.Optional): Extra options to pass to the underlying RPC call.
- `timeout` [(Optional\[float\])](https://docs.python.org/library/typing.html#typing.Optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.

**Returns:**

- None

For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/board/client/index.html#viam.components.board.client.AnalogClient.write).

```python {class="line-numbers linkable-line-numbers"}

> my_board = Board.from_robot(robot=robot, name=”my_board”)

Author: ```suggestion
my_board = Board.from_robot(robot=robot, name="my_board")
```
